src/react-integration/hooks/useInvalidator.ts

```diff
@@ -0,0 +1,28 @@
+import { useContext, useCallback } from 'react';
+
+import { ReadShape, Schema } from '../../resource';
+import { DispatchContext } from '../context';
+
+/** Invalidate a certain item within the cache */
+export default function useInvalidator<
+Params extends Readonly<object>,
+S extends Schema
+>(selectShape: ReadShape<S, Params, any>): (params: Params) => void {
+  const { getUrl } = selectShape;
+  const dispatch = useContext(DispatchContext);
+
+  const invalidateDispatcher = useCallback(
+    (params: Params) => {
```

Comment:
currently params is not typed to be null. Do you want this to be true? (see useCache for example of how nullable works)

Reply:
good catch 🤦‍♂